src/redo_stats.erl

```diff
@@ -0,0 +1,56 @@
+%% Copyright (C) 2015 Heroku, Inc.
+%%
+%% Permission is hereby granted, free of charge, to any person
+%% obtaining a copy of this software and associated documentation
+%% files (the "Software"), to deal in the Software without
+%% restriction, including without limitation the rights to use,
+%% copy, modify, merge, publish, distribute, sublicense, and/or sell
+%% copies of the Software, and to permit persons to whom the
+%% Software is furnished to do so, subject to the following
+%% conditions:
+%%
+%% The above copyright notice and this permission notice shall be
+%% included in all copies or substantial portions of the Software.
+%%
+%% THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+%% EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+%% OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+%% NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+%% HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+%% WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+%% FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+%% OTHER DEALINGS IN THE SOFTWARE.
+-module(redo_stats).
+
+-export([get_stats/1]).
+
+-ifdef(TEST).
+-compile(export_all).
+-endif.
+
+%% Fetch the stats from Redis and return a proplist of values.
+-spec get_stats(atom() | pid()) -> [{term(), binary()}] | {'error', Reason::term()}.
+get_stats(Conn) ->
+    parse_response(redo:cmd(Conn, get_stats_cmd())).
+
+%%--------------------------------------------------------------------
+%% Internal functions
+%%--------------------------------------------------------------------
+
+get_stats_cmd() ->
+    [<<"INFO">>, <<"default">>].
+
+parse_response(Response) ->
+    Lines = re:split(Response, <<"\r\n">>),
```

Comment:
You can avoid the PCRE engine (which asks for a compile step dynamically on the match pattern) by going `binary:split(Response, <<"\r\n">>, [global])` and likely get something more efficient:

```erlang
14> lists:sum([element(1,timer:tc(fun() -> binary:split(<<"# Server\r\nversion:2.8.19\r\nsha1:00000000\r\n">>, <<"\r\n">>, [global]) end)) || _ <- lists:seq(1,10000)]).
86078
15> lists:sum([element(1,timer:tc(fun() -> re:split(<<"# Server\r\nversion:2.8.19\r\nsha1:00000000\r\n">>, <<"\r\n">>) end)) || _ <- lists:seq(1,10000)]).
155792
```